src/components/Navbar.tsx

```diff
@@ -0,0 +1,144 @@
+];
+
+export const Navbar = () => {
+    const { logout } = useAuth();
+
+    const [isMobile, setIsMobile] = useState(window.innerWidth <= 768);
+    const [isMobileMenuOpen, setMobileMenuOpen] = useState(false);
+
+    const updateNavbarState = () => {
+        setIsMobile(window.innerWidth <= 768);
+    };
+
+    useEffect(() => {
+        window.addEventListener("resize", updateNavbarState);
+        return () => {
+            window.removeEventListener("resize", updateNavbarState);
+        };
+    }, []);
+
+    const renderNavItems = (isMobile: boolean) =>
+        navItems.map(({ path, label, Icon }) => (
+            <li
+                key={label}
+                className="p-4 hover:bg-slate-100 duration-300 transition-colors rounded-md w-full hover:text-black"
+            >
+                <Link
+                    to={path}
+                    className="flex items-center justify-start gap-2"
+                >
+                    {isMobile ? (
+                        label
+                    ) : (
+                        <>
+                            <Icon size={32} />
+                            <span className="hidden md:flex">{label}</span>
+                        </>
+                    )}
+                </Link>
+            </li>
+        ));
+
+    return (
+        <>
+            {isMobile ? (
+                <>
+                    <nav className="flex items-center justify-between p-4 text-white border-b-2 border-b-gray-300">
+                        <div className="flex items-center justify-start">
+                            <Link
+                                className="flex gap-4 items-center z-50"
+                                to="/profile"
+                            >
+                                <img
+                                    className="h-12 w-12"
+                                    src="./src/assets/hh-dashboard-logo.svg"
+                                    alt="HawkHacks Logo"
+                                />
+                            </Link>
+                        </div>
+                        <div className="z-50">
+                            <Hamburger
+                                toggled={isMobileMenuOpen}
+                                toggle={setMobileMenuOpen}
+                                size={24}
+                                color="black"
+                                label="Show navigation menu"
+                            />
+                        </div>
+                    </nav>
+
+                    <div
+                        className={`fixed right-0 top-0 z-40 h-full w-full max-w-[65%] p-10 py-24 bg-gray-200 backdrop-blur-xl transition-transform duration-300 ease-in-out ${
+                            isMobileMenuOpen
+                                ? "translate-x-0 opacity-100"
+                                : "translate-x-full opacity-0"
+                        } `}
+                    >
```

Comment:
```suggestion
                        <div
                            className={`fixed right-0 top-0 z-40 h-full w-full max-w-[12.5rem] p-10 py-24 bg-gray-200 backdrop-blur-xl transition-all duration-300 ease-in-out ${
                                isMobileMenuOpen ? "translate-x-0 opacity-100" : "translate-x-full opacity-0"
                            }`}
                        >
```

![image](https://github.com/LaurierHawkHacks/Dashboard/assets/33383085/e4b92d9a-233c-4fd1-81a4-83517718fb23)
Should it use % for the width of the mobile bar? Could look weird near the upper end of mobile sizes.

Also copied over the smooth opening animation to the closing of the menu as well.

Everything else looks awesome 🔥

Reply:
I noticed this too! Your changes make this look wayyy better